org.eclipse.lemminx/src/main/java/com/thaiopensource/relaxng/pattern/CMRelaxNGElementDeclaration.java

```diff
@@ -273,4 +271,49 @@ public String toString() {
 		result.append(" />");
 		return result.toString();
 	}
+
+	/**
+	 * Returns true if the child element is the second choice and false otherwise.
+	 * 
+	 * @param childElement the child element declaration to check.
+	 * @return true if the child element name is the second choice and false
+	 *         otherwise.
+	 */
+
+	private boolean isSecondChoice(CMRelaxNGElementDeclaration childElement) {
+		// return true if the element is the second choice
+		Pattern patternContent = getPattern().getContent();
+		if (patternContent instanceof ChoicePattern
+				&& ((ChoicePattern) patternContent).getOperand2() instanceof ElementPattern) {
+			Pattern childElementContent = childElement.getPattern().getContent();
+			Pattern secondOperandContent = ((ElementPattern) (((ChoicePattern) patternContent).getOperand2()))
+					.getContent();
+			if (childElementContent == secondOperandContent) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	@Override
+	public Set<CMElementDeclaration> getRequiredElements() {
+		Set<CMElementDeclaration> requiredElements = new LinkedHashSet<>();
+		for (CMElementDeclaration element : getElements()) {
+			if (!isOptional(element.getLocalName()) && !isSecondChoice((CMRelaxNGElementDeclaration) element)) {
+				requiredElements.add(element);
+			}
+		}
+		return requiredElements;
+	}
+
+	@Override
+	public Collection<CMElementDeclaration> getElementsToGenerate() {

```

Comment:
Could you please remove this method. We need to invesitgate more how to manage this usecase with XSD, DTD, and RNG.

The code action which will insert all elements will have some bugs but the most important code action is required elements.